packages/studio-ui/src/web/views/FlowBuilder/nodeProps/SkillCallNode.tsx

```diff
@@ -0,0 +1,72 @@
+import { Button, H3, Tag } from '@blueprintjs/core'
+import { lang } from 'botpress/shared'
+import { FlowView, NodeView } from 'common/typings'
+import _ from 'lodash'
+import React, { Component } from 'react'
+
+import { AccessControl } from '~/components/Shared/Utils'
+import EditableInput from '../common/EditableInput'
+import style from './style.scss'
+import TransitionSection from './TransitionSection'
+
+interface Props {
+  flow: FlowView
+  node: NodeView
+  readOnly: boolean
+  subflows: string[]
+  updateNode: (node: Partial<NodeView>) => void
+  requestEditSkill: (id: string) => void
+}
+
+export default class SkillCallNodePropertiesPanel extends Component<Props> {
+  renameNode = (text) => {
+    if (text) {
+      const alreadyExists = this.props.flow.nodes.find((x) => x.name === text)
+      if (!alreadyExists) {
+        this.props.updateNode({ name: text })
+      }
+    }
+  }
+
+  transformText(text) {
+    return text.replace(/[^a-z0-9-_\.]/gi, '_')
```

Comment:
Can you either move the regex into a constant or add a comment that explains what this regex does?

Reply:
I dnd't touch this piece of code yet, I'll eventually get into reviewing this, will fix only then